Skaffold + Kaniko - Not error when kaniko.noPush is true

<!--
  Issues without logs and details are more complicated to fix.
  Please help us by filling the template below!
-->

### Expected behavior
skaffold should not fail when `noPush` is true under kaniko

### Actual behavior
skaffold errors out with this error

```
INFO[0176] Skipping push to container registry due to --no-push flag 
GET https://1111111.dkr.ecr.ap-southeast-2.amazonaws.com/v2/my-image/manifests/<tag>: MANIFEST_UNKNOWN: Requested image not found
Error: Process completed with exit code 1.
```
Kaniko/Skaffold should not be checking whether the image has been pushed or not if `.build.artifacts.[0].kaniko.noPush` is `true`

### Information

- Skaffold version: <!-- run `skaffold version` -->
```
$ skaffold version
v1.35.1
```
- Operating system: <!-- name and version -->
```
Ubuntu 20.04.3 LTS (Focal Fossa)"
```

- Installed via: <!-- skaffold.dev | Google Cloud SDK | Cloud Code | Chocolatey | Snap | Homebrew | ...-->
GitHub Releases

- Contents of skaffold.yaml:

```yaml
---
apiVersion: skaffold/v2beta26
kind: Config
metadata:
  name: build-image
build:
  cluster:
<redacted>
  artifacts:
  - image: 1111111.dkr.ecr.ap-southeast-2.amazonaws.com/my-image
    kaniko:
      dockerfile: ./Dockerfile_vikas027
      noPush: true
```

### Steps to reproduce the behavior

1. `skaffold build`


<!--
The logs from `skaffold ... -vdebug` can be very helpful.
Please remove or elide any confidential information.
-->

--- a/docs-v2/content/en/schemas/v4beta5.json
+++ b/docs-v2/content/en/schemas/v4beta5.json
@@ -2606,12 +2606,6 @@
           "x-intellij-html-description": "to add timestamps to log format.",
           "default": "false"
         },
-        "noPush": {
-          "type": "boolean",
-          "description": "if you only want to build the image, without pushing to a registry.",
-          "x-intellij-html-description": "if you only want to build the image, without pushing to a registry.",
-          "default": "false"
-        },
         "ociLayoutPath": {
           "type": "string",
           "description": "to specify a directory in the container where the OCI image layout of a built image will be placed. This can be used to automatically track the exact image built by kaniko.",
@@ -2719,7 +2713,6 @@
         "cleanup",
         "insecure",
         "insecurePull",
-        "noPush",
         "force",
         "logTimestamp",
         "reproducible",

--- a/pkg/skaffold/build/gcb/kaniko_test.go
+++ b/pkg/skaffold/build/gcb/kaniko_test.go
@@ -195,16 +195,6 @@ func TestKanikoBuildSpec(t *testing.T) {
 			},
 		},
 		{
-			description: "with NoPush",
-			artifact: &latest.KanikoArtifact{
-				DockerfilePath: "Dockerfile",
-				NoPush:         true,
-			},
-			expectedArgs: []string{
-				kaniko.NoPushFlag,
-			},
-		},
-		{
 			description: "with OCILayoutPath",
 			artifact: &latest.KanikoArtifact{
 				DockerfilePath: "Dockerfile",

--- a/pkg/skaffold/build/kaniko/args.go
+++ b/pkg/skaffold/build/kaniko/args.go
@@ -101,10 +101,6 @@ func Args(artifact *latest.KanikoArtifact, tag, context string) ([]string, error
 		args = append(args, LogTimestampFlag)
 	}
 
-	if artifact.NoPush {
-		args = append(args, NoPushFlag)
-	}
-
 	if artifact.OCILayoutPath != "" {
 		args = append(args, OCILayoutFlag, artifact.OCILayoutPath)
 	}

--- a/pkg/skaffold/build/kaniko/args_test.go
+++ b/pkg/skaffold/build/kaniko/args_test.go
@@ -188,17 +188,6 @@ func TestArgs(t *testing.T) {
 			wantErr: false,
 		},
 		{
-			description: "with NoPush",
-			artifact: &latest.KanikoArtifact{
-				DockerfilePath: "dir/Dockerfile",
-				NoPush:         true,
-			},
-			expectedArgs: []string{
-				NoPushFlag,
-			},
-			wantErr: false,
-		},
-		{
 			description: "with OCILayoutPath",
 			artifact: &latest.KanikoArtifact{
 				DockerfilePath: "dir/Dockerfile",

--- a/pkg/skaffold/build/kaniko/types.go
+++ b/pkg/skaffold/build/kaniko/types.go
@@ -53,8 +53,6 @@ const (
 	LogFormatFlag = "--log-format"
 	// LogTimestampFlag additional flag
 	LogTimestampFlag = "--log-timestamp"
-	// NoPushFlag additional flag
-	NoPushFlag = "--no-push"
 	// OCILayoutFlag additional flag
 	OCILayoutFlag = "--oci-layout-path"
 	// RegistryCertificateFlag additional flag

--- a/pkg/skaffold/schema/latest/config.go
+++ b/pkg/skaffold/schema/latest/config.go
@@ -1344,9 +1344,6 @@ type KanikoArtifact struct {
 	// InsecurePull if you want to pull images from a plain HTTP registry.
 	InsecurePull bool `yaml:"insecurePull,omitempty"`
 
-	// NoPush if you only want to build the image, without pushing to a registry.
-	NoPush bool `yaml:"noPush,omitempty"`
-
 	// Force building outside of a container.
 	Force bool `yaml:"force,omitempty"`
 

